### PR TITLE
Fixed issue .ui-state-focus missing from main navigation in 2.4.5 and 2.4.6

### LINKED ISSUE #37300: .ui-state-focus missing from main navigation in 2.4.5 and 2.4.6
### Preconditions and environment

- Magento version: 2.4.5 and 2.4.6

### Steps to reproduce

1. Install a vanilla Magento version 2.4.5 - 2.4.6
2. Open developer tools and inspect a submenu item in the main navigation to watch classes
3. Using keyboard tab to the main menu then use arrow keys to navigate to submenu items

### Expected result

When focus is on a menu item a class should be added of `ui-state-focus`

### Actual result

No `ui-state-focus` class is added, just `ui-state-active`

### Additional information

Accessibility issue introduced in 2.4.5 and remains an issue in 2.4.6. The `ui-state-focus` class is missing on items in the menu in focus preventing styles from being added visually identifying which element is in focus for keyboard only users. This violates [WCAG Success Criteria 2.4.7 Focus Visible (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html). The class exists as expected in version 2.4.3-p2.

### Release note

_No response_

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by rostilos**
### Description (*)

The bug was reproduced because the updated version of jquery-ui menu widget now adds the "ui-state-active" class instead of "ui-state-focus" to the element with focus. When the jquery-ui library was updated (somewhere between version 2.4.3 and 2.4.5) these edits were added but the styles were not changed. 
In this PR, the selectors for the "ui-state-focus" class have been replaced by "ui-state-active" because the "ui-state-focus" class is no longer added to the menu item with focus in the menu functionality of the jquery-ui library widget

### Fixed Issues (if relevant)

1. Fixes magento/magento2#37300

### Manual testing scenarios (*)

1. Install a vanilla Magento version 2.4.5 - 2.4.6
2. Open developer tools and inspect a submenu item in the main navigation to watch classes
3. Using keyboard tab to the main menu then use arrow keys to navigate to submenu items
4. When focus is on a menu/submenu item a class should be added of "ui-state-focus" ( Also the menu item with focus must be visually marked )
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rostilos** [comment]
@magento run all tests

**Comment by rostilos** [comment]
@magento run all tests

**Comment by schizek** [comment]
@rostilos I don't think just styling the `ui-state-active` class will work here since `active` and `focus` have different meanings.  In this case, if you put focus on a child menu item, the parent item is still "active", so the styles would apply to both which will be an issue.  Ideally the `ui-state-focus` class would return so that could be styled separately from `ui-state-active`.

**Comment by engcom-Hotel** [comment]
@magento run all tests

File: lib/web/css/source/lib/_navigation.less [hunk 1]
```diff
@@ -99,7 +99,7 @@
 
             &.active {
                 .all-category {
-                    .ui-state-focus {
+                    .ui-state-active {
```
Comment (engcom-Hotel) [inline lib/web/css/source/lib/_navigation.less:102]: We can change the class name in backward-compatible way like the below:

.ui-state-active, .ui-state-focus

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @rostilos,

Thanks for the contribution!

Please review my comment below.

Thanks

**Comment by engcom-Echo** [comment]
@rostilos 

We have verify PR changes and follow steps for verification mentioned [here](https://github.com/magento/magento2/pull/37571#issue-1738458295). But the PR changes does not fix the issue.

Please have a look at below video and let us know if we have missed anything here

https://github.com/magento/magento2/assets/51681435/afee46ff-3730-499c-94cf-901cf029571a

**Comment by engcom-Echo** [comment]
@rostilos moving it on hold on the basis of above [comment](https://github.com/magento/magento2/pull/37571#issuecomment-1786782904)

**Comment by engcom-Echo** [comment]
@rostilos did you get a chance to have a look on [comment](https://github.com/magento/magento2/pull/37571#issuecomment-1786782904)

**Comment by rostilos** [comment]
@engcom-Echo Yes, I have looked at it, and will try to fix it as soon as possible, can't spare the time right now, but will give an update soon. Thanks

**Comment by engcom-Echo** [comment]
@rostilos did you get a chance to have a look on https://github.com/magento/magento2/pull/37571#issuecomment-1786782904?

**Comment by engcom-Echo** [comment]
As PR need some work to be done based on https://github.com/magento/magento2/pull/37571#issuecomment-1786782904 Moving this PR in draft state.

**Pull request closed without merging**